packages/caliper-core/test/worker/worker-message-handler.js

```diff
@@ -0,0 +1,162 @@
+const PreparedMessage = require('../../lib/common/messages/preparedMessage');
+const TestResultMessage = require('../../lib/common/messages/testResultMessage');
+const WorkerMessageHandler = require('../../lib/worker/worker-message-handler');
+const CaliperWorker = require('../../lib/worker/caliper-worker');
+
+describe('WorkerMessageHandler', () => {
+    let messengerMock, connectorFactoryMock, workerMock, sandbox;
+    beforeEach(() => {
+        sandbox = sinon.createSandbox();
+        messengerMock = {
+            on: sandbox.stub(),
+            send: sandbox.stub(),
+            getUUID: sandbox.stub().returns('worker-uuid')
+        };
+
+        connectorFactoryMock = sandbox.stub();
+
+        workerMock = sinon.createStubInstance(CaliperWorker);
+
+        mockery.enable({
+            warnOnReplace: false,
+            warnOnUnregistered: false,
+            useCleanCache: true
+        });
+
+        mockery.registerMock('./caliper-worker', {
+            CaliperWorker: () => workerMock
+        });
+    });
+
+    afterEach(() => {
+        mockery.deregisterAll();
+        mockery.disable();
+        sandbox.reset();
+    });
+
+    it('should handle "register" message', async () => {
+        const handler = new WorkerMessageHandler(messengerMock, connectorFactoryMock);
+        const message = {
+            getType: () => MessageTypes.Register,
+            getSender: () => 'manager-uuid',
+            stringify: () => 'register message'
+        };
+
+        await handler._handleRegisterMessage(message);
+
+        sinon.assert.calledOnce(messengerMock.send);
+        sinon.assert.calledWith(messengerMock.send, sinon.match.instanceOf(ConnectedMessage));
+    });
+
+    it('should handle "assignId" message', async () => {
+        const handler = new WorkerMessageHandler(messengerMock, connectorFactoryMock);
+        const message = {
+            getType: () => MessageTypes.AssignId,
+            getWorkerIndex: () => 1,
+            stringify: () => 'assignId message'
+        };
+
+        await handler._handleAssignIdMessage(message);
+
+        sinon.assert.calledOnce(messengerMock.send);
+        sinon.assert.calledWith(messengerMock.send, sinon.match.instanceOf(AssignedMessage));
+    });
+
+    it('should handle "initialize" message', async () => {
+        const handler = new WorkerMessageHandler(messengerMock, connectorFactoryMock);
+        connectorFactoryMock.resolves('connector-instance');
+        const message = {
+            getType: () => MessageTypes.Initialize,
+            stringify: () => 'initialize message'
+        };
+
+        await handler._handleInitializeMessage(message);
+
+        sinon.assert.calledOnce(connectorFactoryMock);
+        sinon.assert.calledOnce(messengerMock.send);
+        sinon.assert.calledWith(messengerMock.send, sinon.match.instanceOf(ReadyMessage));
+    });
+
+    it('should handle "prepare" message', async () => {
```

Comment:
We need to describe here the behaviour, just saying handle doesn't provide enough information of the behaviour being tested.